packages/matplotlib/src/html5_canvas_backend.py

```diff
@@ -279,6 +298,98 @@ def draw_image(self, gc, x, y, im, transform=None):
+            self.fontd[key] = font_value
+        font, font_file_name = font_value
+        font.clear()
+        font.set_size(prop.get_size_in_points(), self.dpi)
+        return font, font_file_name
+
+    def get_text_width_height_descent(self, s, prop, ismath):
+        if ismath:
+            image, d = self.mathtext_parser.parse(s, self.dpi, prop)
+            w, h = image.get_width(), image.get_height()
+        else:
+            font, _ = self._get_font(prop)
+            font.set_text(s, 0.0, flags=LOAD_NO_HINTING)
+            w, h = font.get_width_height()
+            w /= 64.0
+            h /= 64.0
+            d = font.get_descent() / 64.0
+        return w, h, d
+
+    def _draw_math_text(self, gc, x, y, s, prop, angle):
+        rgba, descent = self.mathtext_parser.to_rgba(s, gc.get_rgb(), self.dpi,
+                                                     prop.get_size_in_points())
+        height, width, _ = rgba.shape
+        angle = math.radians(angle)
+        if angle != 0:
+            self.ctx.save()
+            self.ctx.translate(x, y)
+            self.ctx.rotate(-angle)
+            self.ctx.translate(-x, -y)
+        self.draw_image(gc, x, -y-descent, np.flipud(rgba))
+        if angle != 0:
+            self.ctx.restore()
+
+    def draw_text(self, gc, x, y, s, prop, angle, ismath=False, mtext=None):
+        def _load_font_into_web(loaded_face):
+            document.fonts.add(loaded_face)
+            window.font_loaded = True
+            _font_set.add(font_face_arguments)
+            self.fig.draw_idle()
+
+        if ismath:
+            self._draw_math_text(gc, x, y, s, prop, angle)
+            return
+        angle = math.radians(angle)
+        width, height, descent = \
+            self.get_text_width_height_descent(s, prop, ismath)
+        x -= math.sin(angle) * descent
+        y -= math.cos(angle) * descent - self.ctx.height
+        font_size = \
+            self.points_to_pixels(prop.get_size_in_points())
+
+        _, font_file_name = self._get_font(prop)
+
+        font_face_arguments = (prop.get_name(), 'url({0})'.format(
+                               _base_fonts_url+font_file_name))
+
+        if font_face_arguments not in _font_set:
+            f = FontFace.new(*font_face_arguments)
+            f.load().then(_load_font_into_web)
+
+        font_property_string = '{0} {1} {2:.3g}px {3}, {4}'.format(
+            prop.get_style(),
+            prop.get_weight(),
+            font_size, prop.get_name(),
+            prop.get_family()[0]
+        )
+        if angle != 0:
+            self.ctx.save()
+            self.ctx.translate(x, y)
+            self.ctx.rotate(-angle)
+            self.ctx.translate(-x, -y)
+        self.ctx.font = font_property_string
+        self.ctx.fillStyle = self._matplotlib_color_to_CSS(
+            gc.get_rgb(),
+            gc.get_alpha()
+        )
+        self.ctx.fillText(s, x, y)
+        self.ctx.fillStyle = '#000000'
```

Comment:
Do we always need to set the color to black after drawing text?  Maybe that's indicating a bug elsewhere?

Reply:
Not that I can observe now. This basically says that revert back to `default` setting once we are done.

Reply:
I am leaving it for now, but I have noted this for future.